### ISSUE TITLE
Data leakage

### ISSUE THREAD

**Issue by Duongvu05** [git-disl/BERT4ETH#12]
Hi, thank you for sharing this work!

I've been experimenting with your current data splitting approach, and I believe there might be a limitation regarding generalization to new accounts and transactions.

Currently, the dataset is split after padding/truncating sequences to max sequence length. However, this may introduce data leakage or overlap between accounts across the train/test sets, especially in sequential fraud detection tasks. In real-world scenarios, we often need to detect fraud from entirely new accounts with unseen transaction patterns.

In my experiments, when applying the current splitting logic:

    The model tends to overfit on the seen accounts.

    The F1-score for class 1 (fraudulent) is very low — only around 0.1, which indicates poor performance in fraud detection with threshold 0.5.

**Comment by Bayi-Hu** [comment]
Hi. The problem is solved by averaging representations of multiple sequences derived from one sequence during the testing.

**Comment by Bayi-Hu** [comment]
In fraud account detection, we apply train/test split at account level rather than sequence level. See in our code, "embedding_list.append(np.mean(embeds, axis=0))"

`def load_embedding():

    # must have checkpoint
    if FLAGS.init_checkpoint == None:
        raise ValueError("Must need a checkpoint for evaluation")

    checkpoint_name = FLAGS.init_checkpoint.split("/")[0]
    model_index = str(FLAGS.init_checkpoint.split("/")[-1].split("_")[1])
    embeddings = np.load("./inter_data/embedding_" + checkpoint_name + "_" + model_index + ".npy")
    address_for_embedding = np.load("./inter_data/address_" + checkpoint_name + "_" + model_index + ".npy")

    # group by embedding according to address
    address_to_embedding = {}

    for i in range(len(address_for_embedding)):
        address = address_for_embedding[i]
        embedding = embeddings[i]
        try:
            address_to_embedding[address].append(embedding)
        except:
            address_to_embedding[address] = [embedding]

    # group to one
    address_list = []
    embedding_list = []

    for addr, embeds in address_to_embedding.items():
        address_list.append(addr)
        if len(embeds) > 1:
            embedding_list.append(np.mean(embeds, axis=0))
        else:
            embedding_list.append(embeds[0])

    # final embedding table
    X = np.array(np.squeeze(embedding_list))

    return X, address_list`
